.github/workflows/ScreenShotTest.yml

```diff
@@ -43,15 +43,54 @@ jobs:
         if: ${{ hashFiles('preview-screenshots/out/failures/*.png') != '' }}
         uses: actions/upload-artifact@v3
         with:
-          name: scrennshot-test-results
+          name: screenshot-test-results
           path: preview-screenshots/out/failures
 
-      - name: Comment PR
+      - name: New checkout for the companion branch
+        id: checkout_docs
         if: ${{ hashFiles('preview-screenshots/out/failures/*.png') != '' }}
+        uses: actions/checkout@v3
+        with:
+          path: temp_docs
+
+      - name: Commit the screenshot to the branch
+        id: docs_images
+        if: steps.checkout_docs.outcome == 'success'
         continue-on-error: true # https://github.com/DroidKaigi/conference-app-2022/issues/497
+        env:
+          BRANCH_NAME: companion_${{ github.head_ref || github.ref_name }}
+        run: |
+          cd temp_docs/
+          git switch -C $BRANCH_NAME
+          mkdir -p docs/screenshots/
+          cp -a ../preview-screenshots/out/failures/. docs/screenshots/
+          git add .
+          git config --global push.default current
+          git config --global remote.pushDefault origin
+          git commit -m "Upload screenshots to github page."
+          git push -f
+          cd ..
+          cd preview-screenshots/out/failures
+          echo ::set-output name=images::$(ls | jq -R -s -c 'split("\n")[:-1]' | jq -r --arg IMAGE_PATH "https://raw.githubusercontent.com/droidkaigi/conference-app-2022/$BRANCH_NAME/docs/screenshots/" '.[] |= $IMAGE_PATH + .')
+
+      - name: Build PR Comment with Preview
+        id: pr_comment
+        if: steps.docs_images.outcome == 'success'
+        continue-on-error: true # https://github.com/DroidKaigi/conference-app-2022/issues/497
+        env:
+          ALL_SCREENSHOTS: ${{ steps.docs_images.outputs.images }}
+        # Build a comment message with the image from snapshot images for demo purpose.
+        run: |
+          echo "There are differences in Compose previews:" > report.md
+          echo >> report.md # A blank line.
+          echo "$ALL_SCREENSHOTS" | jq -r '.[]' | while read -r image; do 
```

Comment:
It works well!
Can we `grep "delta"`?

https://github.com/DroidKaigi/conference-app-2022/pull/640

Reply:
Ah I see, the diffed image names start with "delta". We can filter it to only include the failed tests. Let me quickly try it.

Reply:
FYI: https://github.com/eneim/conference-app-2022/pull/1#issuecomment-1253412406 will bring the change here in a few seconds.

Reply:
Done: https://github.com/DroidKaigi/conference-app-2022/pull/616/commits/e9b694c4da5a4b3eb2991cd3eec4f716c4bcb004